modules/ingest-common/src/main/java/org/opensearch/ingest/common/URLDecodeProcessor.java

```diff
@@ -48,11 +48,7 @@ public final class URLDecodeProcessor extends AbstractStringProcessor<String> {
     }
 
     public static String apply(String value) {
-        try {
-            return URLDecoder.decode(value, "UTF-8");
```

Comment:
I think there was a reason to throw `IllegalArgumentException` here. What are the consequences on a REST API status code for example (this would need tests)?

Reply:
I don't think so. The `UnsupportedEncodingException` depends on which `URLDecoder.decode()` we use, if you pass the encoding as a string (earlier), then you need to handle the checked exception while if you use the class `CharSet` (now) as the encoding you don't.

Reply:
Yeah, it is no longer possible for this line of code to generate an `UnsupportedEncodingException`.